client/session.go

```diff
@@ -153,19 +153,28 @@ func (s *session) Open() error {
 	s.seriesIteratorsPool.Init()
 	s.Unlock()
 
-	currTopologyMap := s.topo.GetAndSubscribe(s.topoMapCh)
+	currTopologyMap, topologyWatch, err := s.topo.GetAndSubscribe()
+	if err != nil {
+		return err
+	}
 	if err := s.setTopologyMap(currTopologyMap); err != nil {
 		s.Lock()
 		s.state = stateNotOpen
 		s.Unlock()
 		return err
 	}
+	s.topoWatch = topologyWatch
 
 	go func() {
 		log := s.opts.GetInstrumentOptions().GetLogger()
-		for value := range s.topoMapCh {
-			if err := s.setTopologyMap(value); err != nil {
-				log.Errorf("could not update topology map: %v", err)
+		for _ = range s.topoWatch.C() {
+			m, ok := s.topoWatch.Get().(topology.Map)
+			if !ok {
+				log.Errorf("Received invalid update.")
+				continue
+			}
+			if err = s.setTopologyMap(m); err != nil {
+				log.Errorf("Could not update topology map: %v", err)
```

Comment:
Hey for consistency could you make both these log statements start with lower case letters and remove periods at the end?